### PR TITLE
VE-221 Migrate deprecated APIs

### PR DESCRIPTION
**Pull request by m4manjesh**
Closes [VE-221](https://sce.myjetbrains.com/youtrack/issue/VE-221/Migrating-from-deprecated-APIs)

## Changes

During the RUI migration to MUI v6 [here](https://github.com/IPG-Automotive-UK/react-ui/pull/984), we noticed it is good to handle deprecations now before they become breaking changes. This PR focuses on migrating recently deprecated APIs to stay up-to-date and improve the developer experience.

A brief description of what this pull request solves / introduces.

- Many system props such as `mt`, `bgColor` and more are deprecated in Box, Typography, Link, Grid, and Stack components. So here we are moving the most of the of the system props to `sx` props. See below example
Screenshot 2024-11-27 122048

- `Hidden` component was already deprecated. Remove it and use `useMediaQuery` to achieve the same behaviour as follows
Screenshot 2024-11-27 122202

- In [ThemeProvider.tsx](https://github.com/IPG-Automotive-UK/react-ui/pull/989/files#diff-9f7f950eeb101f410b8e0af8bae8386a88a565945932b0fb1ae41875c5ff1d95) i have added `mixins.toolbar.minHeight to 64` to fix `AppLauncherover` overlapping issue with `AppHeader` in VIRTO apps. Also changed using custom `64px` in `VirtoAppLayout` and `AppLayout` to calculate `height` and `top`, instead of using `theme.mixins.toolbar.minHeight`.

Before
Screenshot 2024-11-27 123832

After
Screenshot 2024-11-27 123839

- All the analogous APIs such as `components` and `componentProps`  are deprecated and eventually removed. Using  standardized `slots` and `slotProps` instead
Screenshot 2024-11-27 124200
Screenshot 2024-11-27 124654

## Dependencies

This needs to wait to merge until this [PR ](https://github.com/IPG-Automotive-UK/react-ui/pull/984) gets merged

## UI/UX

There are some small UI changes related to `AppLauncher` top spacing as mentioned above. It would be good to get reviewed by you @Sowbhagya-ipg.

## Testing notes

Most of the components touched and most of the changes are identical in this PR. But still, make sure all stories are still working the same as before. Check that the changes i made in `VirtoAppLayout` and `AppLayout` didn't break the behaviour we have.

## Author checklist

Before I request a review:

- [x] I have reviewed my own code-diff.
- [x] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [ ] ~I have included appropriate tests.~
- [x] I have checked that the Lint and Test workflows pass on Github.
- [ ] ~I have populated the deploy-preview with relevant test data.~
- [x] I have tagged a UI/UX designer for review if this PR includes UI/UX changes.

### CODE REVIEW

**Comment by Sowbhagya-ipg** [comment]
The app layout changes looks good @m4manjesh

**Comment by nikitharani** [comment]
image
Sidebar item title not visible in darkmode

**Review by nikitharani — CHANGES_REQUESTED** [review]
The changes are mostly good, with only a few minor adjustments needed.

File: src/PasswordChangeDialog/PasswordChangeDialog.jsx [hunk 1]
```diff
@@ -132,17 +131,20 @@ export default function PasswordChangeDialog({
...
+                  slotProps={{
+                    input: { "aria-label": "newPasswordRepeat" }
+                  }}
                 />
               </Grid>
             </Grid>
           )}
           {status === "success" && (
-            <Alert severity="success" style={{ marginBottom: 16 }}>
+            <Alert severity="success" sx={{ marginBottom: 1 }}>
```
[the remark this answers is not in the document]
Comment (mattcorner) [reply src/PasswordChangeDialog/PasswordChangeDialog.jsx:142]: In the MUI spacing convention 1 unit is 8px, so the correct replacement would be `sx={{ mb: 2 }}`

**Review by mattcorner — COMMENTED** [review]
Looks good to me. Tested the AppLayout changes and works well. Reran the mui deprecation codemod and picked up no other changes. Just Nikithas feedback to address.

**Comment by m4manjesh** [comment]
> image image
> 
> Sidebar item text not visible in darkmode

Good catch. Seems like this is working fine with `VirtoAppLayout`. But still this is an issue in this PR, we have similar kind of issues with storybook mode switching in few places, this will be addressed with a PR coming for  [VE-252](https://sce.myjetbrains.com/youtrack/issue/VE-252/Combine-dark-and-light-theme-for-MUI-v6)

**Review by nikitharani — CHANGES_REQUESTED** [review]
Please change mb in passwordChangeDialog otherwise everything lgtm.

**Pull request merged**